docs/src/misc.md

```diff
@@ -177,3 +177,135 @@ AbstractAlgebra.LowercaseOff
 AbstractAlgebra.terse
 AbstractAlgebra.is_terse
 ```
+
+## Linear solving interface for developers
+
+AbstractAlgebra has a generic interface for linear solving and we describe here
+how one may extend this interface. For the user-facing functionality of linear
+solving, see [Linear Solving](@ref solving_chapter).
+
+### Matrix normal forms
+
+To distinguish between different algorithms, we use type traits of abstract type
+`MatrixNormalFormTrait` which usually correspond to a certain matrix normal
```

Comment:
I think most of the traits are only known in the `Solve` submodule, so it would be good to either add `Solve.` in front of all mentions or add an explaining sentence.

Reply:
I added a line. Maybe we could also get rid of `Solve` at some point...